Sources/Sentry/SentryProfiler.mm

```diff
@@ -306,126 +349,97 @@ + (void)startWithHub:(SentryHub *)hub
-        return nil;
-    }
-
     return serializedProfileData([_gCurrentProfiler._state copyProfilingData], transaction,
-        profileID, profilerTruncationReasonName(_gCurrentProfiler->_truncationReason),
-        _gCurrentProfiler -> _hub.scope.environmentString
+        self.profileId, profilerTruncationReasonName(_gCurrentProfiler->_truncationReason),
+        _gCurrentProfiler->_hub.scope.environmentString
             ?: _gCurrentProfiler->_hub.getClient.options.environment,
         _gCurrentProfiler->_hub.getClient.options.releaseName,
         [_gCurrentProfiler->_metricProfiler serializeForTransaction:transaction],
-        [_gCurrentProfiler->_debugImageProvider getDebugImagesCrashed:NO],
-        _gCurrentProfiler -> _hub);
+        [_gCurrentProfiler->_debugImageProvider getDebugImagesCrashed:NO], _gCurrentProfiler -> _hub
+#    if SENTRY_HAS_UIKIT
+        ,
+        self._screenFrameData
+#    endif // SENTRY_HAS_UIKIT
+    );
 }
 
-+ (void)timeoutAbort
+- (void)timeoutAbort
 {
-    std::lock_guard<std::mutex> l(_gProfilerLock);
-
-    if (!_gCurrentProfiler) {
-        SENTRY_LOG_WARN(@"No current global profiler manager to stop.");
-        return;
-    }
-    if (![_gCurrentProfiler isRunning]) {
+    if (![self isRunning]) {
         SENTRY_LOG_WARN(@"Current profiler is not running.");
         return;
     }
 
     SENTRY_LOG_DEBUG(@"Stopping profiler %@ due to timeout.", _gCurrentProfiler);
-    [self stopProfilerForReason:SentryProfilerTruncationReasonTimeout];
+    [self stopForReason:SentryProfilerTruncationReasonTimeout];
 }
 
-+ (void)backgroundAbort
+- (void)backgroundAbort
 {
-    std::lock_guard<std::mutex> l(_gProfilerLock);
-
-    if (!_gCurrentProfiler) {
-        SENTRY_LOG_WARN(@"No current global profiler manager to stop.");
-        return;
-    }
-    if (![_gCurrentProfiler isRunning]) {
+    if (![self isRunning]) {
         SENTRY_LOG_WARN(@"Current profiler is not running.");
         return;
     }
 
     SENTRY_LOG_DEBUG(@"Stopping profiler %@ due to timeout.", _gCurrentProfiler);
-    [self stopProfilerForReason:SentryProfilerTruncationReasonAppMovedToBackground];
+    [self stopForReason:SentryProfilerTruncationReasonAppMovedToBackground];
 }
 
-+ (void)stopProfilerForReason:(SentryProfilerTruncationReason)reason
+- (void)stopForReason:(SentryProfilerTruncationReason)reason
 {
-    [_gCurrentProfiler->_timeoutTimer invalidate];
-    [_gCurrentProfiler stop];
-    _gCurrentProfiler->_truncationReason = reason;
-#    if SENTRY_HAS_UIKIT
-    [SentryDependencyContainer.sharedInstance.framesTracker resetProfilingTimestamps];
-#    endif // SENTRY_HAS_UIKIT
+    [_timeoutTimer invalidate];
+    [_metricProfiler stop];
+    _truncationReason = reason;
+
+    if (![self isRunning]) {
+        SENTRY_LOG_WARN(@"Profiler is not currently running.");
+        return;
+    }
+
+    _profiler->stopSampling();
+    SENTRY_LOG_DEBUG(@"Stopped profiler %@.", self);
```

Comment:
Converted the class methods to instance methods, which simplifies their implementations a bit: no more locking or nil checks required: `+[timeoutAbort]`, `+[backgroundAbort]` and `+[stopProfilerForReason]`.